packages/noya-state/src/reducers/themeReducer.ts

```diff
@@ -666,6 +668,52 @@ export function themeReducer(
         draft.selectedSymbolGroup = '';
       });
     }
+    case 'addGradientAsset': {
+      const [, name, value] = action;
+
+      return produce(state, (draft) => {
+        const assets = draft.sketch.document.assets;
+
+        const newAsset: Sketch.GradientAsset = {
+          _class: 'MSImmutableGradientAsset',
+          do_objectID: uuid(),
+          name: name,
+          gradient: value,
+        };
+
+        assets.gradientAssets.push(newAsset);
+        assets.gradients.push(value);
+      });
+    }
+    case 'removeGradientAsset': {
+      const [, id] = action;
+      const ids = typeof id === 'string' ? [id] : id;
+
+      return produce(state, (draft) => {
+        const assets = draft.sketch.document.assets;
+
+        const indexes = assets.gradientAssets.flatMap((g, index) =>
+          ids.includes(g.do_objectID) ? [index] : [],
+        );
+
+        indexes.reverse().forEach((index) => {
+          assets.gradients.splice(index, 1);
+          assets.gradientAssets.splice(index, 1);
+        });
+      });
+    }
+    case 'setGradientAssetName': {
+      const [, id, name] = action;
+      const ids = typeof id === 'string' ? [id] : id;
+
+      return produce(state, (draft) => {
+        const gradientAssets = draft.sketch.document.assets.gradientAssets;
+
+        gradientAssets.forEach((g) => {
+          if (ids.includes(g.do_objectID)) g.name = name;
```

Comment:
Do we want to update the name in `assets.gradients` here too? E.g. using the `indexes` approach above

Reply:
the gradients don't have names, only the gradientAssets